src/map/skill.c

```diff
@@ -19180,6 +18967,1095 @@ bool skill_parse_row_changematerialdb(char* split[], int columns, int current) {
+	} else if (sk->max > MAX_SKILL_LEVEL || sk->max <= 0) {
+		ShowError("skill_validate_skilldb: Invalid Max Level %d specified for skill Id %d in '%s', skipping...\n", sk->max, sk->nameid, source);
+		return false;
+	}
+	
+	memmove(&skill->dbs->db[idx],sk,sizeof(struct s_skill_db));
+	strdb_iput(skill->name2id_db, skill->dbs->db[idx].name, skill->dbs->db[idx].nameid);
+	script->set_constant2(skill->dbs->db[idx].name, (int)skill->dbs->db[idx].nameid, false, false);
+	
+	return true;
+}
+	
+/**
+ * Reads skill_db.conf from relative filepath and processes [ Smokexyz/Hercules ]
+ * entries into the skill database.
+ * @param  filename     contains the file path and name.
+ * @return boolean      true on success
+ */
+bool skill_read_skilldb(const char *filename)
+{
+	struct config_t skilldb;
+	struct config_setting_t *sk, *conf;
+	char filepath[256];
+	int count=0, index=0;
+	bool duplicate[MAX_SKILL] = {0};
+	
+	nullpo_retr(false, filename);
+
+	sprintf(filepath,"db/%s",filename);
+	
+	if (!libconfig->load_file(&skilldb, filepath)) {
+		return false; // Libconfig error report.
+	} else if ((sk=libconfig->setting_get_member(skilldb.root, "skill_db")) == NULL) {
+		ShowError("skill_read_skilldb: Skill DB could not be loaded, please check '%s'.\n", filepath);
+		return false;
+	}
+	
+	while ((conf = libconfig->setting_get_elem(sk,index++))){
+		int idx=0, skill_id=0, temp=0, check=0;
+		const char *tstr;
+		struct config_setting_t *t = { 0 }, *tt = { 0 };
+		struct s_skill_db tmp_db = { 0 };
+		bool type=false;
+		
+		/* Skill ID */
+		if (!libconfig->setting_lookup_int(conf, "Id", &skill_id)) {
+			ShowError("skill_read_skilldb: Skill Id not specified for entry %d in '%s', skipping...\n", index+1, filepath );
+			continue;
+		}
+		
+		tmp_db.nameid = skill_id;
+		
+		if((idx = skill->get_index(skill_id)) == 0)
+			continue;
+
+		if (duplicate[idx] == true) {
+			ShowWarning("skill_read_skilldb: Duplicate Skill Id %d in entry %d in '%s', skipping...\n", skill_id, index+1, filepath);
+			continue;
+		}
+		/* Skill Name Constant */
+		if (!libconfig->setting_lookup_string(conf, "Name", &tstr)) {
+			ShowError("skill_read_skilldb: Name not specified for skill Id %d in '%s', skipping...\n", skill_id, filepath);
+			continue;
+		} else {
+			char *name = aStrdup(tstr);
+			safestrncpy(tmp_db.name, trim(name), sizeof(tmp_db.name));
+			aFree(name);
+		}
+		/* Skill Description */
+		if (libconfig->setting_lookup_string(conf, "Description", &tstr)) {
+			char *desc = aStrdup(tstr);
+			safestrncpy(tmp_db.desc, trim(desc), sizeof(tmp_db.desc));
+			aFree(desc);
+		}
+		/* Max Level */
+		if (!libconfig->setting_lookup_int(conf, "MaxLevel", &temp)) {
+			ShowError("skill_read_skilldb: MaxLevel not specified for skill Id %d in '%s', skipping...\n", skill_id, filepath);
+			continue;
+		} else {
+			tmp_db.max = temp;
+		}
+		/* Range */
+		if ((t=libconfig->setting_get_member(conf, "Range")))
+			skill->config_set_level(t, tmp_db.range);
+		/* Hit */
```

Comment:
Hit type probably better detect in separate function. This need for plugins to override.